kumascript/src/live-sample.js

```diff
@@ -129,9 +129,19 @@ function getLiveSampleIDs(slug, source) {
   for (let token of tokens) {
     if (
       token.type === "MACRO" &&
-      normalizeMacroName(token.name) === "embedlivesample" &&
-      token.args.length
+      ((normalizeMacroName(token.name) === "embedlivesample" &&
+        token.args.length) ||
+        token.name === "InheritanceDiagram")
     ) {
+      // The InheritanceDiagram is a special macro that, unlike other EmbedLiveSample
+      // macros in that it itself renders the "EmbedLiveSample" with a set of
+      // hardcoded options. The only thing that is variable and comes from the
+      // the raw HTML is the size. So make an exception for this otherwise, the
+      // HTML element ID is lost and this function won't discovered that it is
+      // in fact a live sample here.
+      if (token.name === "InheritanceDiagram") {
+        token.args = ["inheritance_diagram", ...token.args];
+      }
```

Comment:
I think it's better and cleaner to handle this new case on its own:
```js
  for (let token of tokens) {
    if (token.type !== "MACRO") {
      continue;
    }
    const normalizedMacroName = normalizeMacroName(token.name);
    if (normalizedMacroName === "inheritancediagram") {
      result.push(new LiveSampleID("inheritance_diagram", source, token));
    } else if (normalizedMacroName === "embedlivesample" && token.args.length) {
      ...
    }
```

Reply:
Done. r?